src/dotnet/Core/DI/LazyInstance.cs

```diff
@@ -0,0 +1,8 @@
+namespace ActualChat.DI;
+
+public class LazyInstance<T>: Lazy<T> where T : notnull
+{
+    public LazyInstance(IServiceProvider serviceProvider)
```

Comment:
Looks like there is no lazy behaviour. Instance of T is created upon LazyInstance<T> creation.

Reply:
why do you think so? I pass lambda (method ref) creating instance of T to base constructor (of Lazy), not the instance itself

Reply:
Method group confused me.